### PR TITLE
Netcore: Migrated member unit tests into new project/builder pattern

### PR DESCRIPTION
**Pull request by AndyButland**
In this PR I've moved the two test classes `MemberTests` and `MemberGroupTests` into the new unit tests project, and updated them to follow the builder pattern.

In doing so have implemented the necessary builders, including a couple that are generic and can likely be reused in other contexts (`GenericDictionaryBuilder` and `GenericCollectionBuilder`).

I've also created a few more `IWith...` interfaces where it looked like property setting code could be reused.

### CODE REVIEW

File: src/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/Models/MemberTests.cs [hunk 1]
```diff
@@ -15,6 +16,65 @@ public class MemberTests
     {
         private readonly MemberBuilder _builder = new MemberBuilder();
 
+        private const int _testMemberTypeId = 99;
+        private const string _testMemberTypeAlias = "memberType";
+        private const string _testMemberTypeName = "Member Type";
+        private const string _testMemberTypePropertyGroupName = "Content";
+        private const int _testId = 6;
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Tests.UnitTests/Umbraco.Infrastructure/Models/MemberTests.cs]: Have updated.

File: src/Umbraco.Tests.Common/Builders/PropertyGroupBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Umbraco.Core.Models;
+using Umbraco.Tests.Common.Builders.Interfaces;
+
+namespace Umbraco.Tests.Common.Builders
+{
+    public class PropertyGroupBuilder
+        : ChildBuilderBase<MemberTypeBuilder, PropertyGroup>,   // TODO: likely want to genericise this, so can use for document and media types too.
```
[the remark this answers is not in the document]
Comment (AndyButland) [reply src/Umbraco.Tests.Common/Builders/PropertyGroupBuilder.cs]: Updated.

**Comment by AndyButland** [comment]
Thanks for the comments and discussion @elit0451 - have made some updates from your suggestions.

**Comment by bergmania** [comment]
@AndyButland Looks good now :)

I will merge.  One thing we could consider, is to make a method on the `MemberBuilder` to just build a complex object. 
Then this functionality can be reused.

The reason i mention this, is because the tests `MemberTests.Can_Deep_Clone()` and `MemberTests.Can_Serialize_Without_Error()` really dont care about what the object contains, as long as it is complex. We also could consider make a helper method to assert whether two `Member`s are equal.

**Pull request merged**